server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/CompactionExecutorsMetrics.java

```diff
@@ -77,10 +92,47 @@ public synchronized AutoCloseable addExecutor(CompactionExecutorId ceid,
 
   @Override
   public void prepareMetrics() {
+
+    if (externalMetricsSupplier != null) {
+
+      Set<CompactionExecutorId> seenIds = new HashSet<>();
+
+      MetricsRegistry registry = super.getRegistry();
+
+      synchronized (exMetrics) {
+        externalMetricsSupplier.get().forEach(ecm -> {
+          seenIds.add(ecm.ceid);
+
+          ExMetrics exm = exMetrics.computeIfAbsent(ecm.ceid, id -> {
+            ExMetrics m = new ExMetrics();
+            m.queuedGauge = registry.newGauge(id.canonical().replace('.', '_') + "_queued",
```

Comment:
Wondering if a queue being configured out of existence causes any problems for the gauges that may be left around.